### ISSUE TITLE
OrderBy Sqlizer?

### ISSUE THREAD

**Issue by kernle32dll** [Masterminds/squirrel#139]
While converting my existing codebase to squirrel, I stumbled across something odd:

There is no way to use an Sqlizer with the OrderBy of a SelectBuilder. Afaik this should be valid SQL for most database implementations I can think of at the top of my head. To be more concrete, an OrderBy Sqlizer would enable e.g. to use a CaseBuilder for ordering.

I was about to hack together a quick PR, when I realized that this is not as easy to implement on the SelectBuilder side:

My first idea was to adjust the signature of OrderBy from `OrderBy(orderBys ...string)` to `OrderBy(pred interface{}, rest ...interface{})`, which would align with most of the other builder methods. Self-speaking, some adjustments to [selectDatas toSql()](https://github.com/Masterminds/squirrel/blob/b127ed9be03443fe3c0877e391130e3dd3f3107a/select.go#L134) included. However, this would utterly break existing code (especially because the new parameters are just interface{}s, so no compiler warning either). Also, the missing piece is where to put the order direction. Right now, what I have in my mind would look like this:

```go
someSqlizer := ...
sql, args, _ := someSqlizer.ToSql()
selectBuilder.OrderBy(sql + " ASC", args)
```

... which is not very neat. On top of that, one would loose the ability to sort by multiple columns at once, and would have to resort to multiple `OrderBy` calls.

The other option would be the introduction of a new method, which just takes an Sqlizer. This would not break existing code, but would convolute the interface with an additional method for ordering. However, this option also suffers the missing ability to specify the order direction.

So... any thoughts? If a neat solution is decided on, I can probably cook up an PR to implement it.

**Quick after tought**: of course you can do `someSqlizer.ToSql()` and use the raw sql with `OrderBy`. However, this breaks if the sqlizer contains args, which are lost this way. And I really don't want to go the "replace placeholders myself" route.

**Comment by lann** [comment]
I think the best solution would be to do the same as `Join`/`JoinClause`:

1. Change `selectData`s `Joins []string` field to `JoinParts []Sqlizer`
2. Add a new method `OrderByClause(pred interface{}, args ...interface{})` that appends `newPart`s to `JoinParts`
3. Update `OrderBy` to call `OrderByClause` for each argument

**Comment by Kernle32DLL** [comment]
@lann looks fine by me - I'm on vacation right now, but will implement this as soon as I'm back

**Comment by lann** [comment]
Fixed by #178
